src/Symfony/Component/Translation/AltTranslator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Accepts several messages and returns the first translated one in the list.
+ *
+ * A message is considered *not* translated when its translation is the same as its id.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class AltTranslator implements TranslatorInterface
+{
+    private $translator;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param string|string[]|null $id
```

Comment:
why the null case?

Reply:
because it's already allowed in the main translator, there is no reason to be stricter here, especially when that means less code.